Set up Docker & Docker Compose for Bot and MongoDB

cuprum-acid opened the pull request:
I set up Docker, but the original code was failing. That's why I've changed it in some places. Please review and make sure you can start it locally. Also, please check for security issues

cuprum-acid committed:
chore: gitignore

.gitignore file to dodge __pycache__ folders

cuprum-acid committed:
fix(bot): fix code that failed in docker container

../ was forced to hardcode with /app/

cuprum-acid committed:
feat(bot): add docker & docker-compose files

cuprum-acid committed:
feat(documentation): update readme

cuprum-acid commented on docker-compose.yml:
```diff
@@ -0,0 +1,24 @@
+  mongo:
+    image: mongo:latest
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: root
+      MONGO_INITDB_ROOT_PASSWORD: example
+    ports:
+      - "27017:27017"
```
Would like to get some comments about this from @CatOrLeader 

Maybe we can return to it and discuss during CI/CD and Deployment meeting

CatOrLeader replied:
Check the latest version of the compose file I pushed now

cuprum-acid commented on telegram-bot/main.py:
```diff
@@ -29,14 +29,14 @@ async def main():
 
     report_callback_provider = ReportCallbackProvider(
-        path_to_schemas='../common/generated/failures-schemas.json'
+        path_to_schemas='/app/common/generated/failures-schemas.json'
```
Hardcoded, but in Docker container this workdir is called /app

m0t9 replied:
попробуй сделать относительный путь

m0t9 replied:
бот ни про какой app не знает

cuprum-acid replied:
I see the problem. Will try to fix it.

You can help and change path too...

cuprum-acid replied:
The issue was that Docker container did not recognize ../ and I was forced to change it 😄

CatOrLeader replied:
This issue was connected to the **WORKDIR** in the Dockerfile

When you type CMD command, it will execute following bash script inside the folder WORKDIR. When you start the python script, you start it from the folder: `/app`
And when the code see path '../', it go upper than the `/app` folder is placed. Now I fixed it, check

CatOrLeader committed:
fix(dotenv): project

It is important to have exactly single `.env` file in the repository, which will not be pushed to the repo itself

CatOrLeader committed:
fix(dotenv): project

It is important to have exactly single `.env` file in the repository, which will not be pushed to the repo itself

CatOrLeader committed:
Merge remote-tracking branch 'origin/docker/telegram-bot' into docker/telegram-bot

# Conflicts:
#	Dockerfile

CatOrLeader committed:
Merge remote-tracking branch 'refs/remotes/origin/telegram-bot' into docker/telegram-bot

# Conflicts:
#	telegram-bot/.env
#	telegram-bot/reporting/callbacks.py
#	telegram-bot/reporting/reporter.py
